### ISSUE TITLE
MSG_Add misuse in IMGMOUNT program implementation

### ISSUE THREAD

**Issue by FeralChild64** [dosbox-staging/dosbox-staging#2063]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

The IMGMOUNT command implementation (`program_imgmount.cpp`) uses `MSG_Add` wrongly - the implementation looks like this:

```c
	MSG_Add("SHELL_CMD_IMGMOUNT_HELP_LONG",
	        "Mount a CD-ROM, floppy, or disk image to a drive letter.\n"
	      (...)
#if defined(WIN32)
	      (...)
#elif defined(MACOSX)
	      (...)
#else
	      (...)
#endif
```

As a result, the `CONFIG -wl` command will produce different results depending on the OS used to generate language template file, and it is not possible to create a proper translation.

### Steps to reproduce the behaviour.

(discovered during code walktrought)

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by kcgen** [comment]
Good catch, @FeralChild64.

We previously had a handful of OS-specific messages:
 - `Msg_Add("SOME_MSG_WIN32", ...)`
 - `Msg_Add("SOME_MSG_LINUX", ...)`
 - `Msg_Add("SOME_MSG_MACOS", ...)`

We could follow this approach for IMGMOUNT, but it triples the work for translators, and the only differences are in our path examples. 

Given all supported operating systems handle forward-slash path separators, it would be really nice to collapse any remaining multi-OS messages into single "OS-agnostic" messages (and if we need to use paths in example, we do it something like: `"path/to/something"`).

What do you think?

**Comment by FeralChild64** [comment]
OK for me. We should also drop the following examples:

`[color=cyan]/Users/USERNAME/games/doom.iso[reset] -t cdrom` (macOS)
`[color=cyan]/home/USERNAME/games/doom.iso[reset] -t cdrom` (Linux)

as they are OS-specific.

**Comment by Kappa971** [comment]
Windows probably holds 80% of the PC market, changing those strings to `path/to/something` instead of `C:\games\doom.iso` is definitely a downside for Windows users, but I understand that differentiating strings between Windows, Linux and Mac would result in a duplication of `SHELL_CMD_IMGMOUNT_HELP_LONG` so I don't know...

**Comment by kcgen** [comment]
I agree @Kappa971, `path/to/something` is probably still a bit confusing.  When it comes to DOSBox help messages, I'm also not a fan of even using `C:\some\path` because because the HDD mounted inside DOSBox also has a `C:\`, and so requires even more clarification: is this file inside DOSBox or outside DOSbox on the host?

If we look at other cross-platform tools, like `ls` and `cp`, which operate on Linux, BSD, Windows(MSYS/MINGW), macOS, and various other Amiga/MorphOS/etc: they just use `"FILE"` and let the user figure out the path.. be it absolute, relative, or local:

![2022-11-04_11-21](https://user-images.githubusercontent.com/1557255/200071042-8ea04fc4-f6ee-40d7-8dc6-0b0a412fbcfb.png)

![2022-11-04_11-20](https://user-images.githubusercontent.com/1557255/200071049-56d705ac-1713-4730-bd8a-3a97d24d226d.png)

So perhaps `FILE` and `DIRECTORY` are even better.

We can also add clarifications, for example: "Where **FILE** is a relative or absolute path on the host filesystem." (or not.. maybe we say "Where **FILE** is a relative or absolute path inside your DOS mounted drive(s)".

**Comment by Kappa971** [comment]
It might be a good idea, but in DOSBox Staging in the "examples" section there are practical examples for using the command (so [FILE] I don't think it would be good).
A solution could be, if possible, to duplicate only the part relating to the examples.

**Comment by FeralChild64** [comment]
Anyway, I have just started separating the 'core' of the MORE command into a separate class, so that it will be possible to display all the help screens via MORE command, event without TEMP being set - by passing the string to the engine object.

I suggest to wait till I finish, as after that we won't be limited to one screen help texts any more.

**Comment by kcgen** [comment]
I agree @Kappa971 ; I think we can lift out just the example paths (like `C:\CDROM.ISO` vs. `/my/path/cdrom.iso`) and pass those into the common/agnostic help message as a `"%s"` field.

**Comment by dreamer** [comment]
I do not agree with removal of these example lines - they were carefully crafted per-OS to help users understand how to use the command. Even experienced users like me rely on them sometimes 😅.

That being said, I don't know for sure they need to be translated - for example - when using macOS as non-english user, is `/Users/USERNAME` still a valid path or does macOS translate it somehow? (`/Użytkownicy/USERNAME`?).

On Linux, paths to home dir are not translated (paths INSIDE home dir are - based on locale - but we don't care about that).
If both Windows and macOS do not translate paths, then we could just move these few ifdefed lines out of translation altogether and print these few lines raw when needed.

Another option is to move them out of translation, keep only word USERNAME to be translated, and inject them to untranslated ifdefed string as needed

**Comment by johnnovak** [comment]
> I do not agree with removal of these example lines - they were carefully crafted per-OS to help users understand how to use the command. Even experienced users like me rely on them sometimes 😅.

+100, I absolutely dont' want the examples to go, for the same reason.

**Comment by FeralChild64** [comment]
Maybe it would be best to always display both examples? We are no longer limited to one screen of help any more.

**Comment by johnnovak** [comment]
> If both Windows and macOS do not translate paths, then we could just move these few ifdefed lines out of translation altogether and print these few lines raw when needed.

Physical system directory names are never localised on any recent Windows and macOS version. The localisation only happens in File Explorer (Finder on macOS) and file dialogs. The English path names are always used at the filesystem level, without exception, so the localised names are just a presentation layer thing that only happens at the GUI level.

See here:
https://discussions.apple.com/thread/252040
https://stackoverflow.com/questions/11115770/how-does-folder-name-translation-work-in-international-windows-7-versions

**Comment by ftortoriello** [comment]
> I agree @Kappa971 ; I think we can lift out just the example paths (like `C:\CDROM.ISO` vs. `/my/path/cdrom.iso`) and pass those into the common/agnostic help message as a `"%s"` field.

Or maybe do that with only the drive letter/home path and dir separators? That would allow translating strings like `games` in the paths.
Also I think `USERNAME` could be translatable, but `/home/USERNAME` and `/Users/USERNAME` can be replaced by just the `~` character on both Linux and macOS (and `C:` on Windows)

**Comment by johnnovak** [comment]
@FeralChild64 FYI, certain translation strings "flip flopping" is currently a problem with some MIDI config descriptions too (as it's OS dependent).

Let me know what you think of my recent IMGMOUNT help changes:
https://github.com/dosbox-staging/dosbox-staging/pull/3551

**Comment by FeralChild64** [comment]
Already approved the commit - but it doesn't fix this issue.

**Comment by johnnovak** [comment]
Addressed in #3562
